packages/domains-table/src/domains-table/index.tsx

```diff
@@ -114,6 +117,12 @@ export function DomainsTable( {
 		} );
 	}, [ fetchedSiteDomains, domains, sortKey, sortDirection ] );
 
+	const filteredData = useFuzzySearch( {
```

Comment:
Maybe we can filter before sorting, so sorting is applied to a smaller set of data, thus getting faster. For smaller datasets the difference isn't noticeable, but since we're talking about a lot of rows, maybe it can be beneficial.

Reply:
Actually no I just remembered that this doesn't work. Fuzzy searching also does it's own sorting: closer matches get filtered to the top.